### PR TITLE
feat: migrate create user modal into react component

### PR DESCRIPTION
**Pull request by ironAiken2**
### This PR is about migrating user generate modal to React components.

> This PR is the bottom of the stack so that it can be merged into main first.This is why this PR also including the part that will be fixed in #2757 PR.

**Changes:**
- I modified the `UserSettingModal.tsx` component because the input for the `modify_user` query and the `create_user` query are not very different.
- If you don't pass a userEmail value in the web component, the response data for the user query will be null. I took advantage of this and branched the modify and create logic.
- (minor) Modified UserInfoModal to show resource policy and description.

**How to test:**
1. move to `User Credential & Polices` page
2. Verify that the modal looks correctly different on create/modify.
3. Verify that user creation is successful.
4. Verify that the non-required value is mapped and applied correctly.
5. Verify that the User is properly part of the `default`, `model-store` group.
6. Verify that user information is modified correctly. 

|before|after|
|---|---|
|||

|Info Modal||
|---|---|
|||

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by ironAiken2** [comment]
* **#2754** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2754?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ironAiken2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/backend-ai-credential-view.ts [hunk 1]
```diff
@@ -1018,6 +1028,20 @@ export default class BackendAICredentialView extends BackendAIPage {
...
+        ? html`
+            <backend-ai-react-user-setting-dialog
+              value="${JSON.stringify({
+                open: this.openUserSettingModal,
+              })}"
+              @ok="${() => {
+                this.openUserSettingModal = false;
+                this.activeUserList.refresh();
+              }}"
+              @cancel="${() => (this.openUserSettingModal = false)}"
```
Comment (agatha197) [inline src/components/backend-ai-credential-view.ts:1040]: > [!NOTE]  
> This will be moved to react component in #2757

File: react/src/components/UserSettingModal.tsx [hunk 2]
```diff
@@ -172,49 +176,136 @@ const UserSettingModal: React.FC<Props> = ({
...
+            group_ids: default_group_id,
+          };
+          commitCreateUser({
+            variables: {
+              email: values?.email || '',
+              props: props,
+              isNotSupportSudoSessionEnabled: !sudoSessionEnabledSupported,
+              isNotSupportTotp: !isTOTPSupported,
+            },
+            onCompleted: (res) => {
```
Comment (agatha197) [inline react/src/components/UserSettingModal.tsx]: Here too.

File: react/src/components/UserSettingModal.tsx [hunk 3]
```diff
@@ -172,49 +176,136 @@ const UserSettingModal: React.FC<Props> = ({
...
+            (item) => _.includes([undefined, ''], item),
+          );
+          commitModifyUserSetting({
+            variables: {
+              email: values?.email || '',
+              props: props,
+              isNotSupportSudoSessionEnabled: !sudoSessionEnabledSupported,
+              isNotSupportTotp: !isTOTPSupported,
+            },
+            onCompleted: (res) => {
```
Comment (agatha197) [inline react/src/components/UserSettingModal.tsx]:
```suggestion
            onCompleted: (res, errors) => {
```
Let's do more error handling.

File: react/src/components/UserSettingModal.tsx [hunk 4]
```diff
@@ -172,49 +176,136 @@ const UserSettingModal: React.FC<Props> = ({
...
-    });
+  const handleOk = () => {
+    formRef.current
+      ?.validateFields()
+      .then(async (values) => {
+        if (user) {
+          const props: ModifyUserInput = _.omitBy(
+            _.omit(values, ['email', 'password_confirm']),
+            (item) => _.includes([undefined, ''], item),
+          );
```
Comment (agatha197) [inline react/src/components/UserSettingModal.tsx]:
```suggestion
         const props: ModifyUserInput = _.omitBy(
	  _.omit(values, ['email', 'password_confirm']),
	  _.isEmpty
	);
```

File: react/src/components/UserSettingModal.tsx [hunk 5]
```diff
@@ -224,34 +315,60 @@ const UserSettingModal: React.FC<Props> = ({
...
+      destroyOnClose
+      onOk={handleOk}
+      confirmLoading={
+        isInFlightCommitModifyUserSetting || isInFlightCommitCreateUser
+      }
       {...baiModalProps}
     >
       <Form
         ref={formRef}
-        preserve={false}
```
Comment (agatha197) [inline react/src/components/UserSettingModal.tsx]: why did you remove this line?

**Review by agatha197 — DISMISSED** [review]

File: src/components/backend-ai-credential-view.ts (hunk 1 above)
Comment (ironAiken2) [reply src/components/backend-ai-credential-view.ts:1040]: Thanks for the mention. I've add a description of it in the body of the PR.

File: react/src/components/UserSettingModal.tsx (hunk 5 above)
Comment (ironAiken2) [reply react/src/components/UserSettingModal.tsx]: My mistake... just added

File: src/components/backend-ai-credential-view.ts [hunk 6]
```diff
@@ -522,6 +531,7 @@ export default class BackendAICredentialView extends BackendAIPage {
    * @param {EventTarget} list - List webcomponent
    */
   _showList(list) {
+    console.log('list', list);
```
Comment (yomybaby) [inline src/components/backend-ai-credential-view.ts]: Please remove this line.

**Review by yomybaby — CHANGES_REQUESTED** [review]
Can you modify the UserSettingModal to be used as a pure React component without wrapping it in a web component? 💪

**Comment by ironAiken2** [comment]
> Can you modify the UserSettingModal to be used as a pure React component without wrapping it in a web component? 💪 

I addressed this issue in upstack #2757 by changing user list to react component. For the current PR, I chose to focus on adding the ability to select a resource policy first.

**Comment by yomybaby** [comment]
Ok! Let's improve handling suspense to response more faster after user click the button after #2757

**Review by yomybaby — APPROVED** [review]
LGTM
